drivers/sensor/adxl375/adxl375_i2c.c

```diff
@@ -0,0 +1,91 @@
+/* adxl375_i2c.c - I2C routines for ADXL375 driver
+ */
+
+/*
+ * Copyright (c) 2022 Analog Devices
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT adi_adxl375
+
+#include <string.h>
+#include <zephyr/logging/log.h>
+
+#include "adxl375.h"
+
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+
+LOG_MODULE_DECLARE(ADXL375, CONFIG_SENSOR_LOG_LEVEL);
+
+static int adxl375_bus_access(const struct device *dev, uint8_t reg, void *data, size_t length)
+{
+	const struct adxl375_dev_config *config = dev->config;
+
+	if (reg & ADXL375_READ) {
+		return i2c_burst_read_dt(&config->i2c, ADXL375_TO_I2C_REG(reg), (uint8_t *)data,
+					 length);
+	} else {
+		if (length != 1) {
+			return -EINVAL;
+		}
+
+		return i2c_reg_write_byte_dt(&config->i2c, ADXL375_TO_I2C_REG(reg),
+					     *(uint8_t *)data);
+	}
+}
+
+static int adxl375_i2c_reg_read(const struct device *dev, uint8_t reg_addr, uint8_t *reg_data)
+{
+	return adxl375_bus_access(dev, ADXL375_REG_READ(reg_addr), reg_data, 1);
```

Comment:
Couldn't you just use the i2c commands directly in these helper function?

Reply:
Agreed and I can make this change. Just as a heads up though that in the adxl372 and adxl367 code, they do something similar. I was trying my best to keep things consistent.

Reply:
Ah I see, at some point in the past the SPI driver was copied to make the I2C driver.
I don't think you need to keep consistency with the old driver, and it isn't worth making a special change to the older driver unless it is being updated for another reason.